transformer_engine/debug/features/api.py

```diff
@@ -0,0 +1,219 @@
+        return True, processed_config
+
+    def _validate_gemm(self, gemm):
+        assert gemm in ["fprop", "wgrad", "dgrad"], (
+            f"[NVTORCH INSPECT ERROR] Invalid gemm: {gemm}. It must be one of the ['fprop',"
+            " 'wgrad', 'dgrad']."
+        )
+
+    def _process_transformer_engine_config(self, config, **kwargs):
+        """
+        Return config specific to a particular tensor name and gemm that matches the api args.
+        """
+        if "gemms_struct" in config:
+            for cfg in config["gemms_struct"]:
+                self._validate_gemm(cfg["gemm"])
+                if cfg["gemm"] == kwargs["gemm"]:
+                    if kwargs["tensor_parsing"]:
+                        cfg = self._process_tensor_config(cfg, kwargs["tensor_name"])
+                        if not cfg:
+                            return None
+                    cfg_copy = copy.deepcopy(cfg)
+                    config.pop("gemms_struct")
+                    assert (
+                        "enabled" not in cfg_copy
+                    ), "[NVTORCH INSPECT ERROR] Enabled field should not be part of gemms_struct"
+                    config.update(cfg_copy)
+                    return config
+            return None
+        if "gemms" in config:
+            for gemm in config["gemms"]:
+                self._validate_gemm(gemm)
+            if kwargs["gemm"] in config["gemms"]:
+                if kwargs["tensor_parsing"]:
+                    cfg = self._process_tensor_config(config, kwargs["tensor_name"])
+                    if not cfg:
+                        return None
+                config["gemm"] = kwargs["gemm"]
+                config.pop("gemms")
+                return config
+            return None
+        raise ValueError(
+            "[NVTORCH INSPECT ERROR] Provide 'gemms_struct: List[Dict]' or 'gemms: List[str]'"
+            " in the config yaml"
+        )
+
+
+required_kwargs = {
+    "fp8_gemm_enabled": ["gemm"],
+    "modify_tensor_enabled": ["tensor_name", "gemm"],
+    "modify_tensor": ["tensor_name", "gemm"],
+    "inspect_tensor": ["tensor_name"],
+    "inspect_tensor_postquantize": ["tensor_name"],
+    "inspect_tensor_enabled": ["tensor_name"],
+    "inspect_tensor_postquantize_enabled": ["tensor_name"],
+    "default": ["tensor_name", "gemm"],
+}
+
+
+class TEDefaultFeatures:
+    """Transformer Engine API calls default behaviour."""
+
+    def fp8_gemm_enabled(self, *_args, **_kwargs):
+        """API call responsible for choice between high-precision and FP8 GEMM execution."""
+        return True  # if it is false, fp8_gemm will be turned off. Otherwise nothing happens.
+
+    def modify_tensor_enabled(self, *_args, **_kwargs):
+        """API call used to determine whether to run modify_tensor() in the forward."""
+        return False
+
+    def modify_tensor(self, *_args, **_kwargs):
+        """API call used to process the tensor."""
+        raise RuntimeError(
```

Comment:
Not sure if I understand fully this class, but shouldn't it be NotImplementedError instead?